MailCore/Cache/MailboxManager/MailboxManager+Search.swift

```diff
@@ -151,27 +153,29 @@ public extension MailboxManager {
 
             // Update thread in Realm
             for message in filteredMessages {
-                let newMessage = message.detached()
-                newMessage.uid = "offline\(newMessage.uid)"
-                newMessage.fromSearch = true
-
-                let newThread = Thread(
-                    uid: "offlineThread\(message.uid)",
-                    messages: [newMessage],
-                    unseenMessages: 0,
-                    from: Array(message.from.detached()),
-                    to: Array(message.to.detached()),
-                    internalDate: newMessage.internalDate,
-                    date: newMessage.date,
-                    hasAttachments: newMessage.hasAttachments,
-                    hasDrafts: newMessage.isDraft,
-                    flagged: newMessage.flagged,
-                    answered: newMessage.answered,
-                    forwarded: newMessage.forwarded
-                )
-                newThread.makeFromSearch(using: writableRealm)
-                newThread.subject = message.subject
-                searchFolder.threads.insert(newThread)
+                if !(featureAvailableProvider.isAvailable(.emojiReaction) && message.emojiReaction != nil) {
```

Comment:
This can probably be done with an initial predicate
`MailboxManager+Search.swift:115`